articles/service-fabric/service-fabric-cluster-creation-setup-aad.md

```diff
@@ -76,53 +118,222 @@ The script prints the JSON required by the Azure Resource Manager template when
+  -Password 'P@ssword!123' `
+  -Verbose
+```
+#### SetupUser.ps1 admin (read/write) example
+
+```powershell
+.\SetupUser.ps1 -ConfigObj $configobj `
+  -UserName 'TestAdmin' `
+  -Password 'P@ssword!123' `
+  -IsAdmin `
+  -Verbose
+```
+
+### SetupClusterResource.ps1
+
+SetupClusterResource.ps1 can optionally be used to export existing cluster resource ARM template, add / modify 'azureActiveDirectory' configuration, and redeploy template. Use '-Whatif' to only export and modify template but not redeploy configuration change. This script does require Azure 'Az' module and name of the resource group for cluster.
+
+#### SetupClusterResource.ps1 -whatIf example
+
+```powershell
+# requires azure module 'az'
+# install-module az
+$resourceGroupName = 'mysftestcluster'
+.\SetupClusterResource.ps1 -configObj $configObj `
+  -resourceGroupName $resourceGroupName `
+  -WhatIf
+```
+
+Once template has been verified and is ready to process, either rerun script without '-WhatIf' or use powershell commandlet 'New-AzResourceGroupDeployment' to deploy template.
+
+#### SetupClusterResource.ps1 example
+```powershell
+$resourceGroupName = 'mysftestcluster'
+.\SetupClusterResource.ps1 -configObj $configObj `
+  -resourceGroupName $resourceGroupName
+```
+
+> [!NOTE] 
+> Update cluster provisioning ARM templates or scripts with new cluster resource AAD configuration changes.
+
+---
+
+## Granting admin consent
+
+It may be necessary to 'Grant admin consent' for the 'API permissions' being configured. Navigate to [Azure App registrations](https://ms.portal.azure.com/#view/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/~/RegisteredApps) blade and add name of cluster to the filter. For both registrations, open 'API permissions', and select 'Grant admin consent for' if available.
+
+![api-grant](media/service-fabric-cluster-creation-setup-aad/portal-client-api-grant.png)
+
+![api-grant-confirm](media/service-fabric-cluster-creation-setup-aad/portal-client-api-grant-confirm.png)
+
+---
+
+## Verifying Azure AD Configuration
+
+Navigate to the Service Fabric Explorer (SFX) url. This should be the same as parameter webApplicationReplyUrl. An Azure authentication dialog should be displayed. Log on with an account configured with new AAD configuration. Verify administrator account has read / write access and user has read access. Any modification to the cluster, for example performing an action is an administrative action.
+
+
 ## Troubleshooting help in setting up Azure Active Directory
-Setting up Azure AD and using it can be challenging, so here are some pointers on what you can do to debug the issue.
 
-### Service Fabric Explorer prompts you to select a certificate
-#### Problem
+Setting up Azure AD and using it can be challenging, so here are some pointers on what you can do to debug the issue. Powershell transcript logging can be enabled by using the '-logFile' argument on 'SetupApplications.ps1' and 'SetupUser.ps1' scripts to review output.
```

Comment:
```suggestion
Setting up Azure AD and using it can be challenging, so here are some pointers on what you can do to debug the issue. PowerShell transcript logging can be enabled by using the '-logFile' argument on 'SetupApplications.ps1' and 'SetupUser.ps1' scripts to review output.
```